### PR TITLE
Add a flag to disable Flink in the operator

### PR DESCRIPTION
**Pull request by andreaTP**

### CODE REVIEW

File: tools/cloudflow-operator/src/main/scala/cloudflow/operator/action/CloudflowStatus.scala [hunk 1]
```diff
@@ -47,7 +47,7 @@ object CloudflowStatus {
   }
 
   private def podReady(ps: App.PodStatus) = {
-    ps.status == PodStatus.Running && ps.nrOfContainersReady == ps.nrOfContainers && ps.nrOfContainers > 0
+    (ps.status == PodStatus.Running && ps.nrOfContainersReady == ps.nrOfContainers && ps.nrOfContainers > 0) || (ps.nrOfContainers == 0 && ps.nrOfContainersReady == 0)
```
Comment (RayRoestenburg) [inline tools/cloudflow-operator/src/main/scala/cloudflow/operator/action/CloudflowStatus.scala:50]: This does not seem right..

File: tools/cloudflow-operator/src/main/scala/cloudflow/operator/action/CloudflowStatus.scala [hunk 2]
```diff
@@ -57,7 +57,7 @@ object CloudflowStatus {
       nrOfContainersReady: Int,
       nrOfContainers: Int): App.PodStatus = {
     val ready =
-      if (nrOfContainersReady == nrOfContainers && nrOfContainers > 0) PodStatus.ReadyTrue else PodStatus.ReadyFalse
+      if ((nrOfContainersReady == nrOfContainers && nrOfContainers > 0)) PodStatus.ReadyTrue else PodStatus.ReadyFalse
```
Comment (RayRoestenburg) [inline tools/cloudflow-operator/src/main/scala/cloudflow/operator/action/CloudflowStatus.scala:60]: not needed?

File: tools/cloudflow-operator/src/main/scala/cloudflow/operator/action/CloudflowStatus.scala [hunk 3]
```diff
@@ -170,11 +175,25 @@ object CloudflowStatus {
...
-        streamletName = deployment.streamletName,
-        expectedPodCount = Some(expectedPodCount),
-        podStatuses = Nil)
+      if (deployment.runtime == FlinkRunner.Runtime && !runners.contains(FlinkRunner.Runtime)) {
+        App.StreamletStatus(
+          streamletName = deployment.streamletName,
+          expectedPodCount = Some(0),
+          podStatuses = Seq(
+            App.PodStatus(
+              name = "<external>",
```
Comment (RayRoestenburg) [inline tools/cloudflow-operator/src/main/scala/cloudflow/operator/action/CloudflowStatus.scala:184]: why not depend on this `"<external>"` string to identify the podstatus that should be treated differently?

File: tools/cloudflow-operator/src/main/scala/cloudflow/operator/action/CloudflowStatus.scala (hunk 1 above)
Comment (RayRoestenburg) [reply tools/cloudflow-operator/src/main/scala/cloudflow/operator/action/CloudflowStatus.scala:50]: IIRC the nrOfContainers can be 0 at first. If this is just for flink status when flink is disabled, rather use the "<external>" marker string to decide this.

**Review by RayRoestenburg — CHANGES_REQUESTED** [review]
We need to discuss the ready logic.

File: tools/cloudflow-operator/src/main/scala/cloudflow/operator/action/CloudflowStatus.scala [hunk 4]
```diff
@@ -102,8 +102,10 @@ object CloudflowStatus {
       nrOfContainers = nrOfContainers)
   }
 
-  private def getStatusFromContainerStates(containerStates: List[ContainerState], nrOfContainers: Int): String =
-    if (containerStates.nonEmpty) {
+  private def getStatusFromContainerStates(containerStates: List[ContainerState], nrOfContainers: Int): String = {
+    if (nrOfContainers == 0) {
```
Comment (RayRoestenburg) [inline tools/cloudflow-operator/src/main/scala/cloudflow/operator/action/CloudflowStatus.scala:106]: We would prefer to show as pending? not sure about this change.

File: tools/cloudflow-operator/src/main/scala/cloudflow/operator/action/CloudflowStatus.scala (hunk 1 above)
Comment (andreaTP) [reply tools/cloudflow-operator/src/main/scala/cloudflow/operator/action/CloudflowStatus.scala:50]: fixed

File: tools/cloudflow-operator/src/main/scala/cloudflow/operator/action/CloudflowStatus.scala (hunk 2 above)
Comment (andreaTP) [reply tools/cloudflow-operator/src/main/scala/cloudflow/operator/action/CloudflowStatus.scala:60]: yes

File: tools/cloudflow-operator/src/main/scala/cloudflow/operator/action/CloudflowStatus.scala (hunk 4 above)
Comment (andreaTP) [reply tools/cloudflow-operator/src/main/scala/cloudflow/operator/action/CloudflowStatus.scala]: reverted

**Comment by andreaTP** [comment]
should be good to go now @RayRoestenburg

**Review by RayRoestenburg — APPROVED** [review]

**Pull request merged**